js/modules/k6/execution/execution.go

```diff
@@ -227,6 +241,48 @@ func newInfoObj(rt *goja.Runtime, props map[string]func() interface{}) (*goja.Ob
 	return o, nil
 }
 
+// optionsAsObject maps the lib.Options struct that contains the consolidated
+// and deriverd options configuration in a goja.Object.
+//
+// The Go primitive types when they are not set then the Go default type is returned.
+// Objects and Array when they are not set then null is returned.
+func optionsAsObject(rt *goja.Runtime, options lib.Options) (*goja.Object, error) {
+	b, err := json.Marshal(options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode the lib.Options as json: %w", err)
+	}
+
+	var props map[string]interface{}
+	err = json.Unmarshal(b, &props)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode as a map the encoded lib.Options: %w", err)
+	}
+
+	delete(props, "vus")
+	delete(props, "iterations")
+	delete(props, "duration")
+	delete(props, "stages")
+
+	obj := rt.NewObject()
+	mustSetReadOnlyProperty := func(k string, v interface{}) {
+		err := obj.DefineDataProperty(k, rt.ToValue(v), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE)
+		if err != nil {
+			common.Throw(rt, err)
+		}
+	}
+
+	for kprop, vprop := range props {
+		mustSetReadOnlyProperty(kprop, vprop)
+	}
+
+	mustSetReadOnlyProperty("consoleOutput", options.ConsoleOutput)
+
+	// TODO: implement the MarshalText function
```

Comment:
```suggestion
	// TODO: expose after implementing a `NullIPPool.MarshalText` function
```
I believe that it nice to mention where we have to implement the `MarshalText`

And maybe just implement it in a separate commit? :smile:

Reply:
The latest commit includes the NullIPPool MarshalText